test/distributed/test_c10d_nccl.py

```diff
@@ -2648,6 +2648,47 @@ def test_nccl_non_blocking_wait_with_barrier(self):
                 "TORCH_NCCL_ASYNC_ERROR_HANDLING"
             ] = prev_nccl_async_error_handling
 
+    @requires_nccl()
+    @requires_nccl_version((2, 4, 0), "Need NCCL 2.4+ for error checking")
+    @skip_if_lt_x_gpu(3)
+    def test_get_future_result(self):
+        def print_fut(fut):
+            print("future is ready:")
+            print(fut.value())
```

Comment:
ideally we can use a callback function to manipulate some values so that we can verify the time when it is called rather than just do the printing.

Reply:
@pytorchbot merge

Reply:
Will update in the stacked PR: https://github.com/pytorch/pytorch/pull/138001